### PR TITLE
feat(insights): Warn about WAU/MAU in total value trends

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

ZEN-11743 showcases how confusing and prone to errors the combination "Weekly/Monthly active users" math with total value trends is. For example, when you have a date range that's "Last 90 days", what does "Monthly active users" mean? There really is no intuitively correct answer that I can think of, so we should at the very least discourage this combination.

## Changes

Added a warning for WAU/MAU series on total value trends:

warn

## How did you test this code?

Added an E2E test.

### CODE REVIEW

File: frontend/src/types.ts [hunk 1]
```diff
@@ -1836,14 +1836,19 @@ export interface DatedAnnotationType extends Omit<AnnotationType, 'date_marker'>
...
+    ActionsLineGraphCumulative = 'ActionsLineGraphCumulative',
+    BoldNumber = 'BoldNumber',
     ActionsPie = 'ActionsPie',
-    ActionsBar = 'ActionsBar',
     ActionsBarValue = 'ActionsBarValue',
+    ActionsTable = 'ActionsTable',
     WorldMap = 'WorldMap',
-    BoldNumber = 'BoldNumber',
+}
+export enum ChartDisplayCategory {
```
Comment (Twixes) [inline frontend/src/types.ts:1848]: Added `ChartDisplayCategory` as an abstraction over the specific display types

I also wanted to use it in the backend, the way we reuse `ChartDisplayType` thanks to `schema.{ts,json,py}`, but unfortunately `ts-json-schema-generator` doesn't support re-exporting, so we can't do that (unless we start using `ChartDisplayCategory` in a type defined in `schema.ts` itself). I put out a [fix for the library](https://github.com/vega/ts-json-schema-generator/issues/567) (interesting opportunity to learn more about TS syntax internals), but in the meantime it's fine for this to be frontend-only

Comment (Twixes) [reply frontend/src/types.ts:1848]: `ChartDisplayCategory` would specifically be super useful for caching

**Comment by thmsobrmlr** [comment]
Might it be better if we hide these two items for total value insights? Or other way around: When would I use the combination of these two? Right now I find the warning a bit strongly worded and would go for something like "in most cases you want to use unique users, only for ... it makes sense to use wau/mau".

File: frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx [hunk 2]
```diff
@@ -549,19 +557,33 @@ function useMathSelectorOptions({
...
+                icon: shouldWarnAboutTrailingMath ? <IconWarning /> : undefined,
+                label: definition.name,
+                tooltip: !shouldWarnAboutTrailingMath ? (
+                    definition.description
+                ) : (
+                    <>
+                        <p>{definition.description}</p>
+                        <i>
+                            In total value insights, it's usually not clear what date range "{definition.name}" refers
+                            to. For full clarity, we recommend using "Unique users" here instead
```
Comment (Twixes) [inline frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:580]: I reworded this a bit to hopefully suggest "Unique users" more gently @thmsobrmlr The tricky bit is that it's really hard to understate how confusing MAU/WAU is in this situation 😅

Comment (thmsobrmlr) [inline frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:580]:
```suggestion
                            to. For full clarity, we recommend using "Unique users" here instead.
```

**Comment by thmsobrmlr** [comment]
> Hmm, hiding is the trickier option, because people switch between time series and total value all the time (like the user in the ticket), so the solution wouldn't be complete without accounting for that workflow. So either:
> 
> Option 1: We only hide MAU/WAU on total insights if not already used – but then we still need the warning to inform about the problem
> 
> Option 2: We switch from MAU/WAU to "Unique users" _for_ the user – only the results would still be unexpected, because there's the same problem: what date range should MAU/WAU _mean_ in a total value context? I think the only solution that isn't surprising would be a modal on display type change to "Hey, what date range do you want for Unique users specifically?" But this also breaks down when there are multiple series where not all are MAU/WAU.
> 
> Hence leaning towards just the warning for now.

Yup, that makes sense. One thought I had: We could have a general warning area of some form for insights, since people might easily miss the warning in the dropdown when switching between time series and total value insights. We could even add an `ignoredWarning` prop to the query nodes so that warnings can be dismissed as well. Nothing for now of course, just an idea since cases like this come up quite frequently. 

> I think what'd be amazing (once HogQL trends are the only trends) would be a series-level rolling average option to replace both WAU/MAU and the current "smoothing" feature. Basically this: #11716

Oh yeah, that looks good. Didn't know about this suggestion at all.

**Review by thmsobrmlr — APPROVED** [review]

**Pull request merged**